components/compositing/touch.rs

```diff
@@ -81,13 +96,38 @@ impl TouchHandler {
 
         self.state = match self.state {
             Nothing => WaitingForScript,
-            Touching | Panning => Pinching,
+            Flinging { .. } => Touching,
+            Touching | Panning { .. } => Pinching,
             WaitingForScript => WaitingForScript,
             DefaultPrevented => DefaultPrevented,
             Pinching | MultiTouch => MultiTouch,
         };
     }
 
+    pub fn on_vsync(&mut self) -> Option<FlingAction> {
+        if let Flinging {
+            velocity,
+            ref cursor,
+        } = &mut self.state
```

Comment:
It think an early return would be a bit nicer here to avoid so much horizontal identation:

```suggestion
        if let Flinging {
            velocity,
            ref cursor,
        } = &mut self.state else {
            return None;
        }
```

Reply:
yeah, thats much nicer